extension/sidebar-pane.js

```diff
@@ -16,148 +16,112 @@ async function _update() {
 
 async function _updateSelectedNode(selectedNode) {
   const issueListEl = document.querySelector("#selected ul");
+  issueListEl.innerHTML = "";
 
-  if (!_isValidElement(selectedNode)) {
-    _render([], issueListEl);
-    return;
-  }
-
-  const issues = [];
-
-  const { actorID, attributes, nodeName } = selectedNode;
-  issues.push(
-    ..._webcompat.getHTMLElementIssues(nodeName, attributes, _targetBrowsers));
-
-  const declarationBlocks =
-    await browser.experiments.inspectedNode.getStyle(actorID, true);
-  for (const declarations of declarationBlocks) {
-    issues.push(
-      ..._webcompat.getCSSDeclarationBlockIssues(declarations, _targetBrowsers));
-  }
-
-  _render(issues, issueListEl);
+  _asyncRenderNodes(0, [selectedNode], true, issueListEl, []).then(() => {
+    if (!issueListEl.querySelector("li")) {
+      _renderNoIssue(issueListEl);
+    }
+  });
 }
 
 async function _updateSubtree(selectedNode) {
   const subtreeEl = document.getElementById("subtree");
   const issueListEl = subtreeEl.querySelector("ul");
 
-  if (!_isValidElement(selectedNode)) {
-    _render([], issueListEl);
-    return;
-  }
-
-  const issues = [];
-
   subtreeEl.classList.add("processing");
   const progressEl = subtreeEl.querySelector("aside label");
 
   progressEl.textContent = "Getting all descendants of the selected node";
   const nodesInSubtree = await browser.experiments.inspectedNode.getNodesInSubtree();
 
-  progressEl.textContent = "Getting web compatibility issues for HTML element";
-  for (const node of nodesInSubtree) {
-    if (!_isValidElement(node)) {
-      continue
+  progressEl.textContent = "Getting web compatibility issues";
+  issueListEl.innerHTML = "";
+  _asyncRenderNodes(0, nodesInSubtree, false, issueListEl, []).then(() => {
+    subtreeEl.classList.remove("processing");
+    if (!issueListEl.querySelector("li")) {
+      _renderNoIssue(issueListEl);
     }
+  });
+}
 
-    const { attributes, nodeName } = node;
-    issues.push(
-      ...
-       _webcompat.getHTMLElementIssues(nodeName, attributes, _targetBrowsers)
-                 .map(issue => {
-                   issue.node = node;
-                   return issue;
-                 })
-    )
+async function _asyncRenderNodes(index, nodes, skipPseudo, listEl, groupsCache) {
```

Comment:
I think renaming this function to something like `_recursiveSubTreeIssuesRendering` might help make it more easy to understand.

Also, please add a short doc comment before this function to explain what it does.

Reply:
Okay, thank you so much.
Since we use for `selectedNode` as well, let me rename to `_recursiveNodesIssuesRendering`.